internal/clients/connect/client_connect.go

```diff
@@ -486,3 +486,46 @@ func (c *ConnectClient) ValidateDeployment(contentID types.ContentID, log loggin
 	}
 	return err
 }
+
+// Handle preflight agent error details
+func (c *ConnectClient) preflightAgentError(agenterr *types.AgentError, contentID types.ContentID) *types.AgentError {
+	agenterr.Code = events.DeploymentFailedCode
+
+	if _, isNotFound := http_client.IsHTTPAgentErrorStatusOf(agenterr, http.StatusNotFound); isNotFound {
+		agenterr.Message = fmt.Sprintf("Cannot deploy content: ID %s - Content cannot be found", contentID)
+	} else if _, isForbidden := http_client.IsHTTPAgentErrorStatusOf(agenterr, http.StatusForbidden); isForbidden {
+		agenterr.Message = fmt.Sprintf("Cannot deploy content: ID %s - You may need to request collaborator permissions or verify the credentials in use", contentID)
+	} else {
+		agenterr.Message = fmt.Sprintf("Cannot deploy content: ID %s - Unknown error: %s", contentID, agenterr.Error())
+	}
+	return agenterr
+}
+
```

Comment:
The new method below, while renamed, is pretty much still the same implementation that was moved from the publish.go file.